File: python/231_Power_of_Two.py
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        # power of two only have one 1 in binary system
        return n > 0 and n & (n - 1) == 0


def test():
    s = Solution()
    print s.isPowerOfTwo(8);


if __name__ == '__main__':
    test()

File: C#/231_Power_of_Two.cs
using System;

namespace LeetCode
{
    public class Solution
    {
        public bool IsPowerOfTwo(int n)
        {
            return n > 0 && (n & (n - 1)) == 0;
        }
        static void Main(string[] args)
        {
            Solution s = new Solution();
            Console.Write(s.IsPowerOfTwo(8));
        }
    }
}
